src/auspex/experiment.py

```diff
@@ -637,17 +637,25 @@ def shutdown(self):
         # This includes stopping the flow of data, and must be done before terminating nodes
         self.shutdown_instruments()
 
-        try:
-            while True:
-                if any([n.is_alive() for n in self.other_nodes]):
-                    logger.warning("Filter pipeline appears stuck. Use keyboard interrupt to terminate.")
-                    time.sleep(2.0)
-                else:
-                    break
-        except KeyboardInterrupt as e:
+        # try:
+        #     while True:
+        #         if any([n.is_alive() for n in self.other_nodes]):
+        #             logger.warning("Filter pipeline appears stuck. Use keyboard interrupt to terminate.")
+        #             time.sleep(2.0)
+        #         else:
+        #             break
+        # except KeyboardInterrupt as e:
+        #     for n in self.other_nodes:
+        #         n.terminate()
+
+        time.sleep(2)
+        if any([n.is_alive() for n in self.other_nodes]):
+            logger.warning('Filter pipeline stuck, terminating.')
```

Comment:
So no more keyboard interrupt? Any reasoning here?

Reply:
Also why not check, or have some small initial sleep before sleeping a full two seconds...

Reply:
I removed the keyboard interrupt and made it raise an exception instead so that the exception could be caught in notebook cells and re-run the experiment automatically. This was mainly so that when many experiments were run in a loop overnight, if one fails, it can be instantly retried which seemed to work. And I chose two seconds to match the digitizer socket timeout which tends to occur at the end of experiments, but this certainly could be refined.